modules/cms/classes/Theme.php

```diff
@@ -471,55 +523,52 @@ public function writeConfig($values = [], $overwrite = false)
     /**
      * Returns the theme preview image URL.
      * If the image file doesn't exist returns the placeholder image URL.
-     * @return string Returns the image URL.
      */
-    public function getPreviewImageUrl()
+    public function getPreviewImageUrl(): string
     {
         $previewPath = $this->getConfigValue('previewImage', 'assets/images/theme-preview.png');
 
-        if (File::exists($this->getPath().'/'.$previewPath)) {
-            return Url::asset('themes/'.$this->getDirName().'/'.$previewPath);
+        if (File::exists($this->getPath() . '/' . $previewPath)) {
+            return Url::asset('themes/' . $this->getDirName() . '/' . $previewPath);
         }
 
         return Url::asset('modules/cms/assets/images/default-theme-preview.png');
     }
 
     /**
      * Resets any memory or cache involved with the active or edit theme.
-     * @return void
      */
-    public static function resetCache()
+    public static function resetCache(bool $memoryOnly = false): void
     {
         self::$activeThemeCache = false;
         self::$editThemeCache = false;
 
-        Cache::forget(self::ACTIVE_KEY);
-        Cache::forget(self::EDIT_KEY);
+        if (!$memoryOnly) {
```

Comment:
```suggestion
        // Sometimes it may be desired to only clear the local cache of the active / edit themes instead of the persistent cache
        if (!$memoryOnly) {
```